pennylane/math/quantum.py

```diff
@@ -972,6 +975,101 @@ def sqrt_matrix(density_matrix):
     return vecs @ qml.math.diag(qml.math.sqrt(evs)) @ qml.math.conj(qml.math.transpose(vecs))
 
 
+def sqrt_matrix_sparse(density_matrix):
+    r"""Compute the square root matrix of a sparse density matrix where :math:`\rho = \sqrt{\rho} \times \sqrt{\rho}`
+
+    Args:
+        density_matrix (sparse): 2D sparse density matrix of the quantum system.
+
+    Returns:
+        (sparse): Square root of the density matrix. Even though data type as `csr_matrix` or `csc_matrix`, the output matrix is not guaranteed to be sparse as well.
+    """
+    if not issparse(density_matrix):
+        raise TypeError(
+            "Only use this method for sparse matrices, or there will be an inevitable performance hit and divergence risk."

```

Comment:
I think it is less common to explain _why_ an error is thrown from a dev perspective :D
I'm not necessarily against it, but it is unconventional I'd say. Maybe
 ```suggestion
            f"sqrt_matrix_sparse only supports sparse matrices, but got {type(density_matrix)}. "
            "There will be an inevitable performance hit and risk of divergence for non-sparse matrices."

```